lib/request.js

```diff
@@ -149,6 +150,14 @@ Object.defineProperties(Request.prototype, {
       return this.raw.url
     }
   },
+  originalUrl: {
+    get () {
+      if (!this[kRequestOriginalUrl]) {
+        this[kRequestOriginalUrl] = this.raw.url
+      }
```

Comment:
Based on the change you add on the line [lib/route.js#472](https://github.com/fastify/fastify/pull/4758/files#diff-cf6996cafd1df0bc168a5c5f3a7470f11fc3aea9f908091d5dd6be2e7b30a2c3R472), this if seems redundant, wdyt?

Reply:
I think that would be required because I am taking the `originalUrl` if defined otherwise, the `url` from the incoming http message. The test fails as well
```ts
 FAIL ​ test/url-rewriting.test.js
 ✖ should be equal

  test/url-rewriting.test.js
  117 |     url: '/',
  118 |     handler: (req, reply) => {
> 119 |       t.equal(req.originalUrl, '/this-would-404-without-url-rewrite')
      | --------^
  120 |       reply.send({ hello: 'world' })
  121 |     }
  122 |   })

  --- expected
  +++ actual
  @@ -1,1 +1,1 @@
  -/this-would-404-without-url-rewrite
  +/

  test: test/url-rewriting.test.js Should rewrite url but keep originalUrl unchanged
  stack: |
    Object.handler (test/url-rewriting.test.js:119:9)
    preHandlerCallback (lib/handleRequest.js:128:37)
    preValidationCallback (lib/handleRequest.js:112:5)
    handler (lib/handleRequest.js:76:7)
    handleRequest (lib/handleRequest.js:24:5)
    runPreParsing (lib/route.js:571:5)
    Object.routeHandler [as handler] (lib/route.js:490:7)
    Router.callHandler (node_modules/find-my-way/index.js:398:14)
    Router.routing [as lookup] (node_modules/find-my-way/index.js:376:17)
    Server.preRouting (fastify.js:810:14)

​ FAIL ​ test/url-rewriting.test.js 1 failed of 21 179.65ms
 ✖ should be equal
```

Reply:
As the `this.raw` contains both, `url` and `originalUrl`. Wdyt if we validate if `this[kRequestOriginalUrl]` exists, and if not we just assign it, and then return it?

Reply:
You mean, remove [Line 472](https://github.com/fastify/fastify/blob/a19b93c143d862fc645d2ccadac7ad98a569494f/lib/route.js#L472) and there, use
```js
 originalUrl: {
    get () {
      if (!this[kRequestOriginalUrl]) {
        this[kRequestOriginalUrl] = this.raw.originalUrl
      }
      return this[kRequestOriginalUrl]
    }
  },
```
?


I tested this and it fails the test. When I logged it, it says, `this.raw.originalUrl` is null

Reply:
Close, you get it right on removing L472. On top of that we can also do instead:

```js
 originalUrl: {
    get () {
      if (!this[kRequestOriginalUrl]) {
        this[kRequestOriginalUrl] = this.raw.originalUrl || this.raw.url
      }
      return this[kRequestOriginalUrl]
    }
  },
```

And that will cover both